openapi_codec/encode.py

```diff
@@ -22,21 +26,116 @@ def generate_swagger_object(document):
     if parsed_url.scheme:
         swagger['schemes'] = [parsed_url.scheme]
 
-    swagger['paths'] = _get_paths_object(document)
+    if not parsed_url.netloc and not parsed_url.scheme:
+        swagger['host'] = document.url
+
+    swagger['definitions'] = _get_definitions(document)
+    swagger['paths'] = _get_paths_object(document, swagger['definitions'])
 
     return swagger
 
 
+def _get_or_update_definitions(update_def_data, update_def_name, definitions):
+    """
+    Updates definitions with provided data If definition is not present in map, returns found definition
+    data in case definition overlaps with existing one.
+    """
+
+    # Check if there's existing definition with same name or props
+    clashing_def_names = \
+        [d for d in definitions.keys() if d.startswith(update_def_name) or definitions.get(d) == update_def_data]
+
+    for clashing_def_name in clashing_def_names:
+        clash_def_data = definitions.get(clashing_def_name)
+        if clash_def_data == update_def_data:
+            return clash_def_data
+    else:
+        if list(clashing_def_names):
+            rand_part = ''.join([random.choice(string.ascii_letters + string.digits) for _ in range(5)])
+            update_def_name = '{}_{}'.format(update_def_name, rand_part)
+        definitions[update_def_name] = update_def_data
+        return update_def_data
+
+
+def _get_field_definition_data(field_item, defs):
+    """
+    Returns dictionary with field definition data.
+    """
+    definition_data = {
+        'type': 'object',
+        'properties': {}
+    }
+
+    if isinstance(field_item, coreschema.Object):
+        props = field_item.properties
+    elif isinstance(field_item.schema, coreschema.schemas.Array):
+        props = field_item.schema.items.properties
+    else:
+        try:
+            props = field_item.schema.properties
+        except AttributeError:
+            props = OrderedDict()
+
+    for f_name, f_schema in iter(props.items()):
+
+        if _get_field_type(f_schema) is 'object':
+            def_data = _get_or_update_definitions(
+                _get_field_definition_data(f_schema, defs),
+                '{}_def_item'.format(f_schema.name),
```

Comment:
I'm starting to play with this.

It works for a super simple API using nested definitions, but when I tried to apply it to our whole site I'm getting a 500 and the traceback is giving an `AttributeError` here saying the name property doesn't exist. I'll try to look into it later, but just a heads up. (I cloned your branch to [nexleaf/python-openapi-codec](https://github.com/nexleaf/python-openapi-codec) and may contribute that way)

Reply:
Sorry, I'm going to be switching to `drf-yasg` as it already supports nested definitions.

Reply:
thanks for heads up. I'd rather close this one and switch to it either.